packages/gatsby/src/utils/page-ssr-module/entry.ts

```diff
@@ -73,19 +87,35 @@ type MaybePhantomActivity =
 
 const createHeaders = createHeadersMatcher(INLINED_HEADERS_CONFIG)
 
-export async function getData({
-  pathName,
-  graphqlEngine,
-  req,
-  spanContext,
-  telemetryResolverTimings,
-}: {
-  graphqlEngine: GraphQLEngine
+interface IGetDataBaseArgs {
   pathName: string
   req?: Partial<Pick<Request, "query" | "method" | "url" | "headers">>
   spanContext?: Span | SpanContext
   telemetryResolverTimings?: Array<IGraphQLTelemetryRecord>
-}): Promise<ISSRData> {
+}
+
+interface IGetDataEagerEngineArgs extends IGetDataBaseArgs {
+  graphqlEngine: GraphQLEngine
+}
+
+interface IGetDataLazyEngineArgs extends IGetDataBaseArgs {
+  getGraphqlEngine: () => Promise<GraphQLEngine>
+}
+
+type IGetDataArgs = IGetDataEagerEngineArgs | IGetDataLazyEngineArgs
```

Comment:
`getData` is public so had to preserve this function still continue to work with `graphqlEngine` being passed in as well as new way of allowing getting it lazily via `getGraphqlEngine`